lib/generators.nix

```diff
@@ -542,4 +578,37 @@ ${expr "" v}
      mkLuaInline :: String -> AttrSet
   */
   mkLuaInline = expr: { _type = "lua-inline"; inherit expr; };
+
+in
+
+# Everything in this attrset is the public interface of the file.
+{
+  inherit
+    mkDconfKeyValue
+    mkKeyValueDefault
```

Comment:
This could be a problem if we try to render documentation from this file in the future (like with other sublibraries). I don't think nixdoc would look at documentation above `inherit` identifiers.

Not a problem for now though. Overall the pattern of being more explicit about what is being exported sounds good to me.

Reply:
Turns out that `nixdoc` sees through these lines without an issue. Opened https://github.com/nix-community/nixdoc/pull/111 to rework a test to show that's the case.